src/main/java/com/conveyal/analysis/results/MultiOriginAssembler.java

```diff
@@ -159,8 +160,8 @@ public MultiOriginAssembler (RegionalAnalysis regionalAnalysis, Job job, String
                 // It's kind of fragile to read from an external network service here. But this is
                 // only triggered when destinations are freeform, which is an experimental feature.
                 destinationPointSet = PointSetCache.readFreeFormFromFileStore(job.templateTask.grid);
-                if (job.templateTask.recordTimes && !job.templateTask.oneToOne) {
-                    if (nOriginsTotal * destinationPointSet.featureCount() > 1_000_000) {
+                if ((job.templateTask.recordTimes || job.templateTask.includePathResults) && !job.templateTask.oneToOne) {
+                    if (nOriginsTotal * destinationPointSet.featureCount() > 16_000_000) {
```

Comment:
O*D will limit the size of the output CSV, but it won't really limit the number of destinations, In the extreme, 16 origins and 1 million destinations is allowed, which would still produce very large results from the worker (probably all 16 million at once). Maybe we should restrict D in addition to or instead of O*D.

Reply:
Also, the message below does not report the true limit. I'll add a destination limit and some new messages.